examples/large_models/ipex_llm_int8/llm_handler.py

```diff
@@ -0,0 +1,619 @@
+                with torch.no_grad(), torch.cpu.amp.autocast(
+                    enabled=self.amp_enabled,
+                    dtype=self.amp_dtype
+                ):
+                    self_jit = torch.jit.trace(converted_model.eval(), example_inputs, strict=False, check_trace=False)
+                    self_jit = torch.jit.freeze(self_jit.eval())
+
+                    self_jit.save(self.quantized_model_path)
+
+                logger.info("IPEX Smooth Quantization has completed successfully")
+
+            else:
+                # run bf16 model
+                example_inputs = self.get_example_inputs()
+                self.user_model = ipex.llm.optimize(
+                    self.user_model.eval(),
+                    dtype = self.amp_dtype,
+                    inplace=True,
+                    deployment_mode=False,
+                )
+                
+                with torch.no_grad(), torch.cpu.amp.autocast(
+                    enabled=self.amp_enabled,
+                    dtype=self.amp_dtype
+                ):
+                    self_jit = torch.jit.trace(self.user_model.eval(), example_inputs, strict=False, check_trace=False)
+                    self_jit = torch.jit.freeze(self_jit.eval())
+
+                    self_jit.save(self.quantized_model_path)
+
+                logger.info("IPEX bf16 optimization is applied successfully")
+        
+            logger.info("Loading the IPEX quantized model")
+            try:
+                self_jit = torch.jit.load(self.quantized_model_path)
+                self_jit = torch.jit.freeze(self_jit.eval())
+            except Exception as e:
+                logger.error("Error: loading the quantized  model failed.", e)
+                exit(0)
+            
+            setattr(self.user_model, "trace_graph", self_jit)
+            logger.info(f"Successfully loaded the Model {model_name} with Intel® Extension for PyTorch*")
+
+        else:
+            # No optimization is applied, but if amx is enabled, it'll be applied during generation routine
+            logger.warning("No IPEX optimization is applied, Pytorch default autocast will be applied if enabled")
+            
+            
+        # set PAD token
+        if self.tokenizer.pad_token is None:
+            if re.search("qwen", self.user_model.config.architectures[0], re.IGNORECASE):
+                self.tokenizer.pad_token = '<|endoftext|>'
+            else:
+                self.tokenizer.pad_token=self.tokenizer.eos_token
+
+            
+        
+    # Different model need to have their inputs supplied in different order unless we pass dict 
+    # For torchserve sending dict is not always possible
+    # This function reorders the input ids, masks, and kv cache based on models 
+    def get_example_inputs(self):
+        example_inputs = None
+        input_ids = torch.ones(32).to(torch.long)
+        attention_mask = torch.ones(len(input_ids))
+        if self.example_inputs_mode == "MASK_POS_KV":
```

Comment:
nit: code could be shared with collate_batch by moving this into a utility function.

Reply:
Removed this part. Now utilizing the dataloader built with `collate_batch` for generating an example input